include/swift/SILOptimizer/OptimizerBridging.h

```diff
@@ -24,15 +24,6 @@
 
 #include "swift/SILOptimizer/Analysis/BasicCalleeAnalysis.h"
 
-#else // USED_IN_CPP_SOURCE
-
-#ifdef SWIFT_SIL_SILVALUE_H
-#error "should not include swift headers into bridging header"
-#endif
-#ifdef LLVM_SUPPORT_COMPILER_H
-#error "should not include llvm headers into bridging header"
-#endif
-
```

Comment:
@eeckstein Why were these checks needed? AFAICT everything builds if I add the following two includes, which hit the check.

https://github.com/apple/swift/pull/70008/files#diff-270b6019bf8675f1ac9a47a482bcdbd4b0e8773d6487665f8dac16129ed4ccc1R25-R28

Reply:
Because the BridgingHeaders should not include any llvm/swift/C++ headers in pure mode. See https://github.com/apple/swift/pull/69039.

So please don't remove those checks!